### PR TITLE
Add Buildkite CI

### PR DESCRIPTION
**Pull request by ClaroHenrique**
This PR aims to add Buildkite CI support and remove the deprecated Gitlab CI.

### CODE REVIEW

**Comment by juliohm** [comment]
@ClaroHenrique can you comment on how close we are to get a final PR here? Can we merge this to see the build happening on buildkite?

**Comment by ClaroHenrique** [comment]
> @ClaroHenrique can you comment on how close we are to get a final PR here? Can we merge this to see the build happening on buildkite?

The Buildkite pipeline was recently created: https://buildkite.com/julialang/opencl-dot-jl
So we can start developing the tests.

**Comment by juliohm** [comment]
We need to install an OpenCL driver in the buildkite image 👍🏽

File: test/test_platform.jl [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
+            @testset "Platform $(p[:profile])" begin
+                @test p != nothing
+                @test pointer(p) != C_NULL
+                for k in [:profile, :version, :name, :vendor, :extensions]
+                    @test p[k] == cl.info(p, k)
+                end
+                # ignore version tests until OpenCL 3 support is available
+                #v = cl.opencl_version(p)
+                #@test 1 <= v.major <= 2
+                #@test 0 <= v.minor <= 2
```
Comment (juliohm) [inline test/test_platform.jl:14]: Is there a reason why this code block is indented differently? It seems that it only has whitespace changes?

Comment (ClaroHenrique) [reply test/test_platform.jl:14]: I added the `@testset "Platform $(p[:profile])" begin` block to get a better log when a test fail.

Comment (juliohm) [reply test/test_platform.jl:14]: Yes, but I think the indentation was changed accidentally? The diff shows various lines of code that are identical, just shifted to the right. Can you please adjust that?

Comment (juliohm) [reply test/test_platform.jl:14]: Also, I think commenting the test on opencl_version isn't ideal. You can just adapt it and replace the upper bound by 3?

File: test/test_program.jl [hunk 2]
```diff
@@ -43,23 +43,25 @@
...
-                @warn("Skipping OpenCL.Program build for Portable Computing Language Platform")
-                continue
-            end
-            @test prg[:build_status][device] == cl.CL_BUILD_SUCCESS
+                # BUILD_SUCCESS undefined in POCL implementation..
+                if device[:platform][:name] == "Portable Computing Language"
+                    @warn("Skipping OpenCL.Program build for Portable Computing Language Platform")
+                    continue
+                end
+                @test prg[:build_status][device] == cl.CL_BUILD_SUCCESS
```
Comment (juliohm) [inline test/test_program.jl:56]: Can you please double check here if you introduced TABS acccidentally?

File: test/test_platform.jl (hunk 1 above)
Comment (juliohm) [reply test/test_platform.jl:14]: Oh nevermind about the indentation. I just saw that you introduced a new testset.

**Review by juliohm — CHANGES_REQUESTED** [review]
We just need to double check the commented test and the TABs.

File: test/test_platform.jl (hunk 1 above)
Comment (ClaroHenrique) [reply test/test_platform.jl]: > Also, I think commenting the test on opencl_version isn't ideal. You can just adapt it and replace the upper bound by 3?

done

File: test/test_program.jl (hunk 2 above)
Comment (ClaroHenrique) [reply test/test_program.jl:56]: Its ok, this code was also inserted inside a testset block.

**Review by juliohm — APPROVED** [review]

**Pull request merged**